common/src/main/java/com/wynntils/core/net/NetResult.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.net;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.wynntils.core.WynntilsMod;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.http.HttpRequest;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+public abstract class NetResult {
+    // By storing the Future here we assure it will not be GC'ed before we are done with it
```

Comment:
Note that getting the Future from CompletableFuture#supplyAsync is just one of the two pathways for getting it, used when the file is already downloaded. For API calls and not-yet-cached files the Future comes from `HTTP_CLIENT.sendAsync`.

Reply:
Do you have any recommendation for replacement for CompletableFuture#supplyAsync?

Reply:
Gotcha, if that's the case make sure the executor being used by the `HTTP_CLIENT` is not doing the same thing. You could force it to use the Fork/Join pool or just create a fixed thread pool executor (with `Executors#newFixedThreadPool`), the way it works is that it will create all the required threads at the moment the method is called, and leave them parking until a request is queued with `Executor#submit`. Tho still it's very unlikely that the fork/join pool will be unavailable, so if you just want to get this done ASAP, can definitely be skipped!

Reply:
I'd like to do something like that if I should do this really properly. See also https://github.com/Wynntils/Artemis/pull/825#discussion_r1048454216; I would really like to make sure that the "supposedly" async processing happens on another thread, to not bog down the main thread. But I guess it is unlikely to be an actual problem, and not just a theoretical, so I'll do as you say and skip it for now.